test/built-ins/String/prototype/indexOf/position-tointeger.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2017 Josh Wolfe. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-string.prototype.indexof
+description: String.prototype.indexOf type coercion for position parameter
+info: >
+  String.prototype.indexOf ( searchString [ , position ] )
+
+  4. Let pos be ? ToInteger(position).
+
+includes: [typeCoercion.js]
+---*/
+
+getValuesCoercibleToIntegerZero().forEach(function(zero) {
+  assert.sameValue("aaaa".indexOf("aa", zero), 0, "with value " + zero);
+});
```

Comment:
As an alternative, calling `getValuesCoercibleToIntegerZero()` could replace the own forEach call. 

i.e.:

```js
valuesCoercibleToIntegerZero(function(zero) {
  assert.sameValue("aaaa".indexOf("aa", zero), 0, "with value " + zero);
});
```

Reply:
I like the idea of passing in a callback instead of using `forEach`. One problem I had while developing these tests is when there was a failure (due to me misreading the spec), it was difficult to determine which test case was causing the problem. It makes a lot of sense to design the test cases so that stack traces are as useful as possible. It was pretty amusing to try to debug print values like `{valueOf() { throw new MyError(); }}`, because every attempt at stringification would throw. Useful stack traces seem like a much more practical solution to the problem than stringifying the object.

I'm a huge fan of long and clear variable names, but I know I'm an outlier in that regard. One concern here is that the functions are exposed at global scope, so I want their full purpose to be in the variable name. I'm also a bit concerned with publishing the helper functions I wrote in the harness file. So perhaps a solution could be exposing an exports object where the name of the object gives context. Perhaps something like `CoercionUtils.ToIntegerZero`. (Now that I write that, I see it's just as long as the original name. Maybe this isn't the best idea.)